earn/src/components/lend/modal/BorrowModal.tsx

```diff
@@ -0,0 +1,181 @@
+      return { text: 'Confirm', enabled: true };
+    case ConfirmButtonState.LOADING:
+      return { text: 'Loading', enabled: false };
+    case ConfirmButtonState.INSUFFICIENT_ASSET:
+      return { text: 'Insufficient Asset', enabled: false };
+    case ConfirmButtonState.DISABLED:
+    default:
+      return { text: 'Confirm', enabled: false };
+  }
+}
+
+export type BorrowModalProps = {
+  isOpen: boolean;
+  selectedBorrows: BorrowEntry[];
+  selectedCollateral: CollateralEntry;
+  setIsOpen: (isOpen: boolean) => void;
+};
+
+export default function BorrowModal(props: BorrowModalProps) {
+  const { isOpen, selectedBorrows, selectedCollateral, setIsOpen } = props;
+  const [collateralAmountStr, setCollateralAmountStr] = useState<string>('');
+  const [borrowAmountStr, setBorrowAmountStr] = useState<string>('');
+  const { activeChain } = useContext(ChainContext);
+
+  const selectedBorrow = selectedBorrows.find(
+    (borrow) => borrow.collateral.address === selectedCollateral.asset.address
+  );
+
+  const selectedLendingPair = selectedCollateral.matchingPairs.find(
+    (pair) =>
+      pair.token0.address === selectedBorrow?.asset.address || pair.token1.address === selectedBorrow?.asset.address
+  );
+
+  const { data: consultData } = useContractRead({
+    abi: volatilityOracleAbi,
+    address: ALOE_II_ORACLE_ADDRESS[activeChain.id],
+    args: [selectedLendingPair?.uniswapPool || '0x', Q32],
+    functionName: 'consult',
+    enabled: selectedLendingPair !== undefined,
+  });
+
+  const userBalance = GN.fromNumber(selectedCollateral.balance, selectedCollateral.asset.decimals);
+  const collateralAmount = GN.fromDecimalString(collateralAmountStr || '0', selectedCollateral.asset.decimals);
+  const borrowAmount = GN.fromDecimalString(borrowAmountStr || '0', selectedBorrow?.asset.decimals ?? 0);
+
+  const maxBorrowAmount = useMemo(() => {
+    if (consultData === undefined || selectedBorrow === undefined) {
+      return null;
+    }
+    const sqrtPriceX96 = GN.fromBigNumber(consultData?.[1] ?? BigNumber.from('0'), 96, 2);
+    const nSigma = selectedLendingPair?.nSigma ?? 0;
+    const iv = consultData[2].div(1e6).toNumber() / 1e6;
+    let ltv = 1 / ((1 + 1 / ALOE_II_MAX_LEVERAGE + 1 / ALOE_II_LIQUIDATION_INCENTIVE) * Math.exp(nSigma * iv));
+    ltv = Math.max(0.1, Math.min(ltv, 0.9));
+
+    let inTermsOfBorrow = collateralAmount;
+    if (selectedLendingPair?.token0.address === selectedCollateral.asset.address) {
+      inTermsOfBorrow = inTermsOfBorrow
+        .mul(sqrtPriceX96)
+        .mul(sqrtPriceX96)
+        .setResolution(selectedBorrow.asset.decimals);
+    } else {
+      inTermsOfBorrow = inTermsOfBorrow
+        .div(sqrtPriceX96)
+        .div(sqrtPriceX96)
+        .setResolution(selectedBorrow.asset.decimals);
+    }
+    const maxBorrowSupply = GN.fromNumber(selectedBorrow.supply, selectedBorrow.asset.decimals);
+    const maxPotentialBorrow = inTermsOfBorrow.recklessMul(ltv);
```

Comment:
These variable names don't make sense. Maybe `maxBorrowSupplyConstraint` and `maxBorrowHealthConstraint` or the same but "maxBorrowConstrainedBy____"